framework/src/base/MooseApp.C

```diff
@@ -1702,6 +1705,8 @@ MooseApp::executeMeshGenerators()
         return;
     }
   }
+
+  _executing_mesh_generators = false;
```

Comment:
I believe we will never actually get to this line because of the `return` in line 1705. Seems like we should either eliminate that early return or also put `_executing_mesh_generators = false` before the early return

Reply:
right... although this line is not important. I will keep this line because the early return logic does not say we will always hit the early return despite that could be the reality.

Reply:
👍